controllers/operatorpolicy_controller.go

```diff
@@ -108,167 +101,300 @@ func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+
+	if specifiedNS, ok := opGroup["namespace"].(string); ok {
+		ogNamespace = specifiedNS
 	}
 
-	// Will only reach this if in inform mode
-	return reconcile.Result{}, nil
+	name, ok := opGroup["name"].(string)
+	if !ok {
+		return nil, fmt.Errorf("name is required in operatorGroup.spec")
+	}
+
+	spec := new(operatorv1.OperatorGroupSpec)
+
+	err = json.Unmarshal(policy.Spec.OperatorGroup.Raw, spec)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling subscription: %w", err)
+	}
+
+	operatorGroup.ObjectMeta.SetName(name)
+	operatorGroup.ObjectMeta.SetNamespace(ogNamespace)
+	operatorGroup.Spec = *spec
+
+	return operatorGroup, nil
 }
 
-// updatePolicyStatus updates the status of the operatorPolicy.
-// In the future, a condition should be added as well, and this should generate events.
-func (r *OperatorPolicyReconciler) updatePolicyStatus(
-	ctx context.Context,
-	policy *policyv1beta1.OperatorPolicy,
-) error {
-	updatedStatus := policy.Status
+func (r *OperatorPolicyReconciler) handleSubscription(ctx context.Context, policy *policyv1beta1.OperatorPolicy) error {
+	watcher := opPolIdentifier(policy.Namespace, policy.Name)
 
-	err := r.Get(ctx, types.NamespacedName{Namespace: policy.Namespace, Name: policy.Name}, policy)
+	desiredSub, err := buildSubscription(policy)
 	if err != nil {
-		OpLog.Info(fmt.Sprintf("Failed to refresh policy; using previously fetched version: %s", err))
-	} else {
-		policy.Status = updatedStatus
+		return fmt.Errorf("error building subscription: %w", err)
 	}
 
-	err = r.Status().Update(ctx, policy)
+	foundSub, err := r.DynamicWatcher.Get(watcher, subscriptionGVK, desiredSub.Namespace, desiredSub.Name)
 	if err != nil {
-		OpLog.Info(fmt.Sprintf("Failed to update policy status: %s", err))
+		return fmt.Errorf("error listing OperatorGroups: %w", err)
+	}
+
+	if foundSub == nil {
+		// Missing Subscription: report NonCompliance
+		err := r.updateStatus(ctx, policy, missingWantedCond("Subscription"), missingWantedObj(desiredSub))
+		if err != nil {
+			return fmt.Errorf("error updating status for a missing Subscription: %w", err)
+		}
+
+		if policy.Spec.RemediationAction.IsEnforce() {
+			err := r.Create(ctx, desiredSub)
+			if err != nil {
+				return fmt.Errorf("error creating the Subscription: %w", err)
+			}
+
+			desiredSub.SetGroupVersionKind(subscriptionGVK) // Create stripped this information
+
+			// Now it should match, so report Compliance
+			err = r.updateStatus(ctx, policy, createdCond("Subscription"), createdObj(desiredSub))
+			if err != nil {
+				return fmt.Errorf("error updating the status for a created Subscription: %w", err)
+			}
+		}
+	} else {
+		// Subscription found; check if specs match
+		desiredUnstruct, err := runtime.DefaultUnstructuredConverter.ToUnstructured(desiredSub)
+		if err != nil {
+			return fmt.Errorf("error converting desired Subscription to an Unstructured: %w", err)
+		}
+
+		if reflect.DeepEqual(desiredUnstruct["spec"], foundSub.Object["spec"]) {
```

Comment:
Also, are there any `omitempty` that the user could specify in `desiredSub` but not be returned in `foundSub`?

Reply:
Since we are dealing with unstructured here, I think you could move out some of the config-policy-controller code from `checkAndUpdateResource` to do the comparison for both must have and must only have.